qiskit_experiments/library/characterization/zz_ramsey.py

```diff
@@ -0,0 +1,164 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+from typing import List, Optional, Sequence
+import numpy as np
+from qiskit import QuantumCircuit, circuit
+from qiskit.providers.backend import Backend
+from qiskit.test.mock import FakeBackend
+from qiskit.circuit import Parameter
+from qiskit_experiments.framework import BaseExperiment, Options
+from qiskit_experiments.library.characterization.analysis.zz_ramsey_analysis import ZZRamseyAnalysis
+
+
+class ZZRamsey(BaseExperiment):
+    r"""Fine DRAG experiment.
+
+    # section: overview
+
+        :class:
+
+    # section:
+    """
+
+    def __init__(
+        self,
+        qubit: (int, int),
+        delays: Optional[Sequence[float]] = None,
+        backend: Optional[Backend] = None,
+    ):
+        super().__init__(qubits=qubit, analysis=ZZRamseyAnalysis(), backend=backend)
+        self.analysis.set_options(outcome="1")
+
+        if delays is not None:
+            self.set_experiment_options(delays=delays)
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        """Default experiment options.
+        Experiment Options:
+        """
+        options = super()._default_experiment_options()
+        options.delays = np.linspace(1e-6, 10e-6, 101)
+        options.zz_rotations = 20
+
+        return options
+
+    def _set_backend(self, backend: Backend):
+        super()._set_backend(backend)
+        if "granularity" in backend.configuration().timing_constraints:
+            self.granularity = backend.configuration().timing_constraints["granularity"]
```

Comment:
We need to do something like this to account for the backend configuration not having timing_constraints:

https://github.com/Qiskit/qiskit-experiments/blob/e06a78bac1c4b8338528e3e6515f5b55c1745f39/qiskit_experiments/library/characterization/cr_hamiltonian.py#L217-L221